### PR TITLE
fix(query-builder): support more operators in join conditions

### LINKED ISSUE #3412: Unable to use `$in` operator in a join clause
**Describe the bug**
When trying to use $in on a join(), knex complains that `in` is not allowed.  Some googling indicates that one should use `onIn` however this does not work either.

**Stack trace**
```
Uncaught TypeError TypeError: The operator "in" is not permitted
    at operator (/node_modules/knex/lib/formatter/wrappingFormatter.js:132:11)
    at onBasic (/node_modules/knex/lib/query/querycompiler.js:902:7)
    at join (/node_modules/knex/lib/query/querycompiler.js:445:40)
    at <anonymous> (/node_modules/knex/lib/query/querycompiler.js:133:40)
    at select (/node_modules/knex/lib/query/querycompiler.js:132:16)
    at toSQL (/node_modules/knex/lib/query/querycompiler.js:73:29)
    at toSQL (/node_modules/knex/lib/query/querybuilder.js:83:44)
    at Target.toQuery (/node_modules/knex/lib/builder-interface-augmenter.js:9:21)
    at toString (/node_modules/knex/lib/query/querybuilder.js:78:17)
    at eval (repl:1:19)```
```
**To Reproduce**
```
const qb = MikroInstance.get()
            .em.fork()
            .qb(Stream)
            .join('event', 'st', {
                status: {
                    $in: ['foo', 'bar'],
                },
            });

qb.getKnexQuery().toString()
```

**Expected behavior**
Valid SQL is produced

**Additional context**
None

**Versions**

| Dependency | Version |
| - | - |
| node | v14.18.3 |
| typescript | 4.7.4 |
| mikro-orm | 5.3.1 |
| your-driver | Sqlite |

### PR DESCRIPTION
**Pull request by r3tard**
Small fix for joining table using operator `$ne`  (`$eq`) with null value.

### Example

```
em.createQueryBuilder(Table, 't').join('relation', 'r', { 'r.nullableField': { $ne: null } });
```

This should create sql ``` `r`.`nullableField` is not null ```, but created sql was  ``` `r`.`nullableField` != null ```

Closes #3412

### CODE REVIEW

File: packages/knex/src/query/QueryBuilderHelper.ts [hunk 1]
```diff
@@ -459,6 +459,16 @@ export class QueryBuilderHelper {
         continue;
       }
 
+      if (cond[key][op] === null) {
+        if (replacement === QueryOperator.$ne) {
+          clause[m](this.knex.raw(`${this.knex.ref(this.mapper(key))} is not ?`, cond[key][op]));
+          continue;
+        } else if (replacement === QueryOperator.$eq) {
```
Comment (B4nan) [inline packages/knex/src/query/QueryBuilderHelper.ts:466]: there is no need for else branch
```suggestion
        }

        if (replacement === QueryOperator.$eq) {
```

File: packages/knex/src/query/QueryBuilderHelper.ts [hunk 2]
```diff
@@ -459,6 +459,16 @@ export class QueryBuilderHelper {
         continue;
       }
 
+      if (cond[key][op] === null) {
+        if (replacement === QueryOperator.$ne) {
+          clause[m](this.knex.raw(`${this.knex.ref(this.mapper(key))} is not ?`, cond[key][op]));
```
Comment (B4nan) [inline packages/knex/src/query/QueryBuilderHelper.ts:464]: this seems like a wrong place (and wrong way) to fix it, we already have support for the `null` comparison, and it works fine without raw queries

Comment (r3tard) [reply packages/knex/src/query/QueryBuilderHelper.ts:464]: we have full support for the `null` only in `where` clause, in `join on` it works only when using the `null` without operator. if you run changed test on master branch, you'll get error

Comment (B4nan) [reply packages/knex/src/query/QueryBuilderHelper.ts:464]: but the join conditions are processed the very same way as where conditions, or not? 

i will need more time to check this out, maybe you are right

Comment (r3tard) [reply packages/knex/src/query/QueryBuilderHelper.ts:464]: no, function `processObjectSubClause` relates only to join. where conditions are processing by `processObjectSubCondition`

File: packages/knex/src/query/QueryBuilderHelper.ts (hunk 1 above)
Comment (r3tard) [reply packages/knex/src/query/QueryBuilderHelper.ts]: changed it to similar code from `where` conditions

**Comment by r3tard** [comment]
I realized that other operators also doesn't work (`$in` for example)

**Review by B4nan — APPROVED** [review]

**Pull request merged**